dask/dataframe/io/parquet/arrow.py

```diff
@@ -1224,7 +1223,15 @@ def _construct_collection_plan(cls, dataset_info):
         # Get/transate filters
         ds_filters = None
         if filters is not None:
-            ds_filters = pq._filters_to_expression(filters)
+            try:
+                ds_filters = pq._filters_to_expression(filters)
+            except TypeError as exc:
+                if "not iterable" in str(exc):
+                    raise TypeError(
+                        "Value of 'in' filter must be a list, set or tuple."
+                    ) from None
+                else:
+                    raise
```

Comment:
Can we avoid relying on logic around parsing error messages emitted? This can be brittle

Reply:
My solution to this question and the previous question was to have an explicit check for invalid values for `in` filters before filters are first used in this function. I also added tests for single-file and folder-type parquet datasets, since they exercise different code paths.